packages/editor/src/pages/Editor/store/settings/sagas.ts

```diff
@@ -54,28 +63,21 @@ export const verifySettings = parsed => {
 };
 
 function* editSettingsCheckSaga(action: ActionType<typeof settingsActions.editFile>) {
-  if (action.payload.newSettings.trim() === '') {
-    yield put(
-      editorActions.openFile({
-        solutionId: SETTINGS_SOLUTION_ID,
-        fileId: DEFAULT_SETTINGS_FILE_ID,
-      }),
-    );
-    return;
-  }
-
   try {
-    const parsed = JSON.parse(action.payload.newSettings);
-
-    const newSettings = verifySettings(parsed);
-
+    const isEmpty = action.payload.newSettings.trim() === '';
```

Comment:
why is the `isEmpty` needed?

Reply:
You can't safeLoad from an empty string.
And safeDump of an empty object gives you the JSON object `{}`, whereas I just wanted an empty file.
It's a good point, though, I'll add it as a code comment.